docs/architecture/0e-plans.md

```diff
@@ -612,7 +612,50 @@ Note that if we want to be able to remove all error code translations (to save c
 
 ## Changes to compilation options
 
-TODO
+### Strategy for removing a compilation option
+
+#### Types of compilation option removals
+
+There are ways in which a compilation option can be removed from the public interface:
+
+* The option is no longer referenced in the code base. Setting it is harmless.
+* The option is still present in the code base, but now implied by other options. This is notably the case for legacy crypto options that are now always controlled via `PSA_WANT_xxx` (`MBEDTLS_PSA_CRYPTO_CONFIG` always on). Setting it may cause the configuration to be inconsistent, so we should complain if the option is set through a configuration file (`mbedtls/mbedtls_config.h`, `psa/crypto_config.h` or similar) rather than internally (in `*/config_adjust*.h`).
+* The option is still present in the code base and not set automatically, but it is not part of the public interface, and will be removed in 1.x/4.x. This happens when we remove a feature that tests still rely on (for example, directly removing some small elliptic curves ([#8136](https://github.com/Mbed-TLS/mbedtls/issues/8136)) would remove some test coverage). We should complain if the option is set, unless it is set from our own test scripts.
+
+#### How to remove a legacy configuration option
+
+Goal: complain if a user tries to set an option in the configuration file (`mbedtls/mbedtls_config.h`, `psa/crypto_config.h` or similar), for options that existed in 3.x but have become internal in 1.0/4.0.
```

Comment:
What if a user unsets an option? For example, they use the default config file plus a few `#undef`s in `MBEDTLS_USER_CONFIG_FILE`. I think we want to complain there too.

Reply:
Yes, that can be detected too. It's rarer and I'd consider it a nice-to-have.